systemd/systemd.go

```diff
@@ -941,24 +947,34 @@ loop:
 		case <-giveup.C:
 			break loop
 		case <-check.C:
-			bs, err := s.systemctl("show", "--property=ActiveState", serviceName)
-			if err != nil {
-				return err
+			allStopped := true
+			stillRunningServices := []string{}
+		serviceCheck:
+			for i := 0; i < len(serviceNames); i++ {
+				bs, err := s.systemctl("show", "--property=ActiveState", serviceNames[i])
+				if err != nil {
+					return err
+				}
+				if !isStopDone(bs) {
+					stillRunningServices = append(stillRunningServices, serviceNames[i])
+					allStopped = false
+				}
+				if !firstCheck {
+					continue serviceCheck
+				}
```

Comment:
this should probably be moved out of this loop, and placed after the `allStopped` check, also it should be `continue loop` or just drop it along with the `loop` label and wrap s.reporter.Notify() with `if !firstCheck { .. }`. in any case the `serviceCheck` label can be dropped